pylint/checkers/classes/class_checker.py

```diff
@@ -1776,10 +1777,16 @@ def _check_protected_attribute_access(
             Klass.
         """
         attrname = node.attrname
-        if (
-            is_attr_protected(attrname)
-            and attrname not in self.linter.config.exclude_protected
-        ):
+        if is_attr_protected(attrname):
+            if attrname in self.linter.config.exclude_protected:
+                return
+            inferred = safe_infer(node.expr)
```

Comment:
The inference is costly, might want to do that at the end, after other checks like "it's a typing it's ok".

Reply:
Right good point. However we may need to re-think this method. I realise now that currently it is geared towards class-attributes. If we move this new inference logic below [these lines](https://github.com/PyCQA/pylint/pull/8364/files#diff-28b08eca9b507acd5447a6f7107823b8d5e627963cccc780601da4f88f842813R1805-R1807) then the new logic is never called as that `klass` object tends to be `None`.